src/main/java/it/gov/pagopa/standinmanager/service/StationCalcService.java

```diff
@@ -92,53 +95,82 @@ public void runCalculations()
-                                Constants.EVENT_REMOVE_FROM_STANDIN,
-                                String.format(
-                                        "removing station [%s] from standIn stations because %s calls were successful"
-                                                + " in the last %s minutes",
-                                        station, successfulCalls, rangeMinutes));
-
-                        String sendResult =
-                                awsSesClient.sendEmail(
-                                        String.format(
-                                                "[StandInManager][%s] Station [%s] removed from standin", env, station),
-                                        String.format(
-                                                "[StandInManager]Station [%s] has been removed from standin"
-                                                        + "\nbecause [%s] calls were successful in the last %s minutes",
-                                                station, successfulCalls, rangeMinutes));
-                        log.info("email sender: {}", sendResult);
+                    if (insertTime.plus(rangeMinutes, ChronoUnit.MINUTES).isBefore(Instant.now()) && successfulCalls > rangeLimit) {
+                        removeStationAutomatically(station, successfulCalls);
                     }
                 });
     }
+
+    public void removeStationFromStandIn(String station) {
+        List<CosmosStandInStation> stations = cosmosStationRepository.getStation(station);
+
+        if (stations.isEmpty()) {
+            throw new IllegalStateException("Station not found: " + station);
+        }
+
+        removeStationManually(station);
+    }
+
+    private void removeStationManually(String station) {
+        String eventInfo = String.format(
+                "removing station [%s] from standIn stations manually", station);
+
+        String emailBody = String.format(
+                "[StandInManager] Station [%s] has been removed from stand-in manually", station);
+
+        removeStation(station, eventInfo, emailBody);
+    }
+
+    private void removeStationAutomatically(String station, long successfulCalls) {
+        String eventInfo = String.format(
+                "removing station [%s] from standIn stations because %s calls were successful"
+                        + " in the last %s minutes", station, successfulCalls, rangeMinutes);
+
+        String emailBody = String.format(
+                "[StandInManager] Station [%s] has been removed from stand-in"
+                        + "\nbecause [%s] calls were successful in the last %s minutes",
+                station, successfulCalls, rangeMinutes);
+
+        removeStation(station, eventInfo, emailBody);
+    }
+
+    private void removeStation(String station, String eventInfo, String emailBody) {
+        log.info(eventInfo);
+
+        List<CosmosStandInStation> stations = cosmosStationRepository.getStation(station);
```

Comment:
If the flow starts from method `removeStationFromStandIn` this query is done 2 times, can be improved according to you?

Reply:
If I understand correctly, could an object of type CosmosStandInStation be passed?